airflow/triggers/temporal.py

```diff
@@ -54,7 +54,8 @@ async def run(self):
         unexpectedly, or handles a DST change poorly.
         """
         # Sleep an hour at a time while it's more than 2 hours away
-        while (self.moment - timezone.utcnow()).total_hours() > 2:
+        hours_until_moment = (self.moment - timezone.utcnow()).total_seconds() / 3600
```

Comment:
You can't move this out of the loop - it will only evaluate once and then you have the risk of an infinite loop. It needs to evaluate in the loop's guard directly.

Reply:
Thanks. That’s helpful!!